www/book.css

```diff
@@ -213,6 +213,32 @@ code .im { color: #a0a; }
 code .co { color: #C4A484; }
 }
 
+/* Tagging code blocks with where they go */
+pre.example, pre.output, .sourceCode { position: relative; }
+
+pre.example::after, pre.output::after, .sourceCode[data-file]::after {
+    position: absolute; top: 0; right: 0; color: #888;
+    font: bold 83% 'Vollkorn', 'Garamond', serif;
+    padding: 1px 1ex; border: 2px solid #aaa;
+    pointer-events: none; /* So you can select / scroll under tag */
+}
+
+/* Note that ::after:hover is not supported in CSS so this is best we can do */
+pre.example:hover::after, pre.output:hover::after, .sourceCode[data-file]:hover::after {
+    opacity: 30%; /* In case it covers some code */
```

Comment:
How about putting it above the code and making space? I find the hover affect distracting and annoying.

Please test on mobile also.